src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/tests/adapters/TreeListAdapter.test.js

```diff
@@ -0,0 +1,634 @@
+    };
+
+    const data = [
+        test1,
+        test2,
+        test3,
+    ];
+    const schema = {
+        title: {
+            type: 'string',
+            visibility: 'yes',
+            label: 'Title',
+        },
+    };
+    const onItemActivationSpy = jest.fn();
+    const treeListAdapter = mount(
+        <TreeListAdapter
+            data={data}
+            disabledIds={[]}
+            loading={false}
+            schema={schema}
+            selections={[]}
+            onPageChange={jest.fn()}
+            page={1}
+            pageCount={1}
+            onItemActivation={onItemActivationSpy}
+        />
+    );
+
+    // expand the row
+    treeListAdapter.find('Row').at(2).find('span.toggleIcon').simulate('click');
+    expect(onItemActivationSpy).toBeCalledWith(6);
+    expect(treeListAdapter.render()).toMatchSnapshot();
+
+    // close the row
+    treeListAdapter.find('Row').at(2).find('span.toggleIcon').simulate('click');
+    expect(treeListAdapter.render()).toMatchSnapshot();
+});
+
+test('Render data with pencil button when onItemEdit callback is passed', () => {
+    const rowEditClickSpy = jest.fn();
+    const test1 = {
+        data: {
+            id: 2,
+            title: 'Test1',
+            hasChildren: false,
+        },
+        children: [],
+    };
+    const data = [
+        test1,
+    ];
+    const schema = {
+        title: {
+            label: 'Title',
+            type: 'string',
+            visibility: 'no',
+        },
+        description: {
+            label: 'Description',
+            type: 'string',
+            visibility: 'yes',
+        },
+    };
+    const treeListAdapter = render(
+        <TreeListAdapter
+            data={data}
+            disabledIds={[]}
+            loading={false}
+            schema={schema}
+            selections={[]}
+            onPageChange={jest.fn()}
+            page={1}
+            pageCount={1}
+            onItemClick={rowEditClickSpy}
+        />
+    );
+
+    expect(treeListAdapter).toMatchSnapshot();
+});
+
+test('Render data with plus button when onItemEdit callback is passed', () => {
+    const rowEditClickSpy = jest.fn();
+    const test1 = {
+        data: {
+            id: 2,
+            title: 'Test1',
+            hasChildren: false,
+        },
+        children: [],
+    };
+    const data = [
+        test1,
+    ];
+    const schema = {
+        title: {
+            label: 'Title',
+            type: 'string',
+            visibility: 'no',
+        },
+        description: {
+            label: 'Description',
+            type: 'string',
+            visibility: 'yes',
+        },
+    };
+    const treeListAdapter = render(
+        <TreeListAdapter
+            data={data}
+            disabledIds={[]}
+            loading={false}
+            schema={schema}
+            selections={[]}
+            onPageChange={jest.fn()}
+            page={1}
+            pageCount={1}
+            onItemClick={rowEditClickSpy}
```

Comment:
The add button should be only added if the `onItemAdd` callback is passed, shouldn't it?